### PR TITLE
Brake controller scripts refactoring and improvements for mouse control

### PR DESCRIPTION
**Pull request by Sharpe49**
Blueprints:
https://blueprints.launchpad.net/or/+spec/scripts-refactoring
https://blueprints.launchpad.net/or/+spec/brake-controller-scripts-improvements

### CODE REVIEW

File: Source/Orts.Simulation/Common/Scripting/BrakeController.cs [hunk 1]
```diff
@@ -15,124 +15,210 @@
...
+            Host.CurrentNotch = notch;
+        }
+
         /// <summary>
         /// Sets the dynamic brake intervention value
         /// </summary>
-        public Action<float> SetDynamicBrakeIntervention;
+        protected void SetDynamicBrakeIntervention(float value)
+        {
+            // TODO: Set dynamic brake intervention instead of controller position
```
Comment (cesarBLG) [inline Source/Orts.Simulation/Common/Scripting/BrakeController.cs:216]: I fixed this TODO some days ago, so if you had the idea of changing this you don't have to.

Comment (Sharpe49) [reply Source/Orts.Simulation/Common/Scripting/BrakeController.cs:216]: Well, if you have a different code available for this, I can integrate it in this PR too.

Comment (cesarBLG) [reply Source/Orts.Simulation/Common/Scripting/BrakeController.cs:216]: The changes were introduced together with some changes to dynamic brake blending (see #732). Your PR seems harmless, so I think it's better to merge this first, and then I'll rebase #732.

This also conflicts with #713 (it adds a hook to command train brakes from cruise control), which is almost ready except for documentation.

File: Source/Orts.Simulation/Common/Scripting/BrakeController.cs [hunk 2]
```diff
@@ -15,124 +15,210 @@
...
-        public Action<float> SetCurrentValue;
+        protected void SetCurrentValue(float value)
+        {
+            Host.CurrentValue = value;
+        }
+
+        /// <summary>
+        /// Sets the intermediate value of the brake controller lever
+        /// </summary>
+        protected void SetIntermediateValue(float value)
```
Comment (cesarBLG) [inline Source/Orts.Simulation/Common/Scripting/BrakeController.cs:190]: How about using a setter now that it's possible?

**Review by cesarBLG — DISMISSED** [review]

**Comment by Sharpe49** [comment]
Thanks, waiting for blueprint validation.

**Comment by Sharpe49** [comment]
Blueprints have been approved... but I have to rebase. You will have to approve the changes once more.

**Review by cesarBLG — APPROVED** [review]
Okay, I've marked #732 as "Not for unstable" so this PR can stay there for a day or two just in case.

**Pull request merged**